optihood/buildings.py

```diff
@@ -1,392 +1,383 @@
+            if b["active"]:
+                bus = solph.Bus(label=b["label"] + '__' + self.__buildingLabel)
+                self.__nodesList.append(bus)
+                self.__busDict[b["label"] + '__' + self.__buildingLabel] = bus
+
+                if b["excess"]:
+                    self.__nodesList.append(
+                        solph.Sink(
+                            label="excess" + b["label"] + '__' + self.__buildingLabel,
+                            inputs={
+                                self.__busDict[b["label"]+'__'+self.__buildingLabel]: solph.Flow(
+                                    variable_costs=b["excess costs"]*(opt == "costs")  # if opt = "env" variable costs should be zero
+                                )}))
+                # add the excess production cost to self.__costParam
+                self.__costParam["excess" + b["label"]+'__'+self.__buildingLabel] = b["excess costs"]
+
+    def addPV(self, data, data_timeseries, opt):
+        # Create Source objects from table 'commodity sources'
+        for i, s in data.iterrows():
+            if opt == "costs":
+                epc=self._calculateInvest(s)[0]
+                base=self._calculateInvest(s)[1]
+                env_capa=s["impact_cap"] / s["lifetime"]
+                env_flow=s["elec_impact"]
+                varc=0 # variable cost is only passed for environmental optimization if there are emissions per kWh of energy produced from the unit
+
+                envParam = [0, s["elec_impact"], s["impact_cap"] / s["lifetime"]]
+
+            elif opt == "env":
+                epc = s["impact_cap"] / s["lifetime"]
+                base = 0
+                env_capa = s["impact_cap"] / s["lifetime"]
+                env_flow = s["elec_impact"]
+                varc = s["elec_impact"] # variable cost is only passed for environmental optimization if there are emissions per kWh of energy produced from the unit
+
+                envParam = [0, s["elec_impact"], s["impact_cap"] / s["lifetime"]]
+
+            self.__nodesList.append(PV(s["label"], self.__buildingLabel,
+                                       self.__busDict[s["to"] + '__' + self.__buildingLabel],
+                                       s["peripheral_losses"], s["latitude"], s["longitude"],
+                                       s["tilt"], s["efficiency"], s["roof_area"],
+                                       s["zenith_angle"], s["azimuth"],
+                                       data_timeseries['gls'],
+                                       data_timeseries['str.diffus'],
+                                       data_timeseries['tre200h0'], s["capacity_min"], s["capacity_max"],
+                                       epc, base, env_capa, env_flow, varc))
+
+            self.__envParam['electricityProdBus' + '__' + self.__buildingLabel] = envParam
+
+            self.__costParam['electricityProdBus' + '__' + self.__buildingLabel] = [self._calculateInvest(s)[0],

```

Comment:
Could you comment on this change? What is the difference in the calculation of cost and environmental parameters compared to the previous version (along with the reason for change)?

Reply:
@vjacquot : could you please handle this ? I don't have the info

Reply:
In the previous version the cost of electricity import was not handled correctly (I can't explain why). Now it is calculated like the CO2 emissions: it fixed the gap with the objective function